engine/src/main/java/org/terasology/rendering/dag/nodes/BackdropNode.java

```diff
@@ -61,11 +73,44 @@
     private static final int RADIUS = 1024;
 
     private WorldRenderer worldRenderer;
+    private BackdropProvider backdropProvider;
+    private RenderingConfig renderingConfig;
+    private WorldProvider worldProvider;
 
     private int skySphere = -1;
     private SetWireframe wireframeStateChange;
 
+    private Material skyMaterial;
+
+    @SuppressWarnings("FieldCanBeLocal")
+    @Range(min = 1.0f, max = 8192.0f)
+    private float sunExponent = 512.0f;
+    @SuppressWarnings("FieldCanBeLocal")
+    @Range(min = 1.0f, max = 8192.0f)
+    private float moonExponent = 256.0f;
+    @SuppressWarnings("FieldCanBeLocal")
+    @Range(min = 0.0f, max = 10.0f)
+    private float skyDaylightBrightness = 0.6f;
+    @SuppressWarnings("FieldCanBeLocal")
+    @Range(min = 0.0f, max = 10.0f)
+    private float skyNightBrightness = 1.0f;
+
+    @SuppressWarnings("FieldCanBeLocal")
+    private Vector3f sunDirection;
+    @SuppressWarnings("FieldCanBeLocal")
+    private Vector3d zenithColor;
+    @SuppressWarnings("FieldCanBeLocal")
+    private SubmersibleCamera activeCamera;
+    @SuppressWarnings("FieldCanBeLocal")
+    private Vector3f cameraDir;
+    @SuppressWarnings("FieldCanBeLocal")
+    private Vector3f cameraPosition;
```

Comment:
You are technically correct.

However, if I do that, you will, in fact, make me create a reference to a pointer inside the process() method. Since you're the one who stopped me from creating a reference to an FBO inside an update() method that might or might not be called during the course of an entire session, are you sure you want me to create it in update(), once every frame? :P